query-engine/connector-test-kit-rs/query-engine-tests/tests/new/regressions/prisma_5952.rs

```diff
@@ -39,8 +39,8 @@ mod regression {
             .assert_success();
 
         let queries = vec![
-            r#"query {findUniqueArtist(where:{firstName_lastName_birth:{firstName:"Michael",netWorth:"236600000.12409"}}) {firstName netWorth}}"#.to_string(),
-            r#"query {findUniqueArtist(where:{firstName_lastName_birth:{firstName:"George",netWorth:"-0.23660010012409"}}) {firstName netWorth}}"#.to_string(),
+            r#"query {findUniqueArtist(where:{firstName_netWorth:{firstName:"Michael",netWorth:"236600000.12409"}}) {firstName netWorth}}"#.to_string(),
+            r#"query {findUniqueArtist(where:{firstName_netWorth:{firstName:"George",netWorth:"-0.23660010012409"}}) {firstName netWorth}}"#.to_string(),
```

Comment:
This tests was added by https://github.com/prisma/prisma-engines/pull/3210. It started failing in my PR. I'm not sure how it's ever been able to work. The compound field name was clearly wrong since day 1, I don't get it 😅...

Reply:
Ohhh, I get it now. The previous algorithm would not validate the fields against the schema. It would simply check if any of the filters were objects and if so, assume it's a compound unique. In this case, it would take every nested filters individually to put it back in a findMany query. eg:

before:
```ts
where: { firstName_lastName_bith: { firstName: "A", netWorth: 12 } }
```

after:
```ts
where: { firstName: "A", netWorth: 12 }
```

Which essentially means the compound unique field name was never really sent to the schema validation, hence why it worked.

With this new PR though, we use the query schema to more safely find out what kind of filters we're dealing. Hence why it's now failing, because it cannot find `firstName_lastName_birth` in the query schema.